### ISSUE TITLE
Currency Conversion for Donation Data from Stripe/Paypal to SFDC

### ISSUE THREAD

**Issue by desiree-mofo** [mozilla/donate.mozilla.org#1977]
Donation amount and currency are passed through to SFDC, but are not converted to USD - which makes analysis difficult because there's no way to convert the values from the currency in which the donation was made into a standardized currency. 

This will impact and will need input from Juan, Will E, Steph, PMAC, Cade, Desiree, Andrew M and MoFo Accounting staff before final decisions can be made as to the requirements, design and final solution of how to handle currencies and currency calculations for Mozilla Foundation donations.

**Comment by desiree-mofo** [comment]
Stripe likely converts the value to USD when the donation occurs, so we should be able to pull that value in to the integration and have a loosely valid value w/out having to do the conversion ourselves.

Paypal may offer similar functionality, but we may have to beef up the integration in order to get this in to SFDC. 

We also need to capture the actual original donation amount and currency in the system so that this information can be used to populate a document that can be sent as a receipt

**Comment by desiree-mofo** [comment]
We will add new fields to Salesforce to capture net amount, fee amount and conversion amount from both Paypal and Stripe. The payment system extracts should be updated to pull these values from Stripe and Paypal and pass them through Basket into SFDC. All 3 new fields will be Currency fields. Desiree to add them to the ADVO sandbox and notify Cade and PMAC of the API field names.

**Comment by desiree-mofo** [comment]
@cadecairos  did some research from Stripe. Notes on what is available: 

> looks like a `charge` object has a related `balance_transaction` object that describes the total net amount after conversion to USD. I'm not sure of the timing of it - it might not be available upon the charge succeeding, but we can test that when the time comes.  if I compare `created` on the charge and the balance transaction, they're identical. so we may only need to expand the charge to grab that balance_transaction data when we receive a webhook.

![image](https://user-images.githubusercontent.com/27289391/37979188-f0a8ab80-31b5-11e8-9c56-68ec130d7500.png)

**Comment by desiree-mofo** [comment]
@cadecairos followed up with:
> yeah, so when stripe sends us a webhook about a new charge, we get `"balance_transaction": "txn_1CAJiYG8Mmx3htnxYPZscAUD"` - so we need to make a Stripe API call to send us the balance transaction
> 
> I'm pretty sure we already "expand" the customer object when we get a charge, so we just need to add balance transaction to the list of expanded objects we want.
> TLDR yes
